TrackableEntities.EF.Core/DbContextExtensions.cs

```diff
@@ -174,9 +179,10 @@ public static async Task LoadRelatedEntitiesAsync(this DbContext context, ITrack
             {
                 if (n.Entry.State == EntityState.Detached)
                     n.Entry.State = EntityState.Unchanged;
+
                 foreach (var reference in n.Entry.References)
                 {
-                    if (!reference.IsLoaded)
```

Comment:
I'm not too sure about this one.

The test data objects were populated and then the reference properties cleared (except for the customer property?). 

After the upgrade, the unit tests were failing on line 181 above as it tried to set the Customer entry state to Unchanged. This was because the Customer had just been loaded into the context through the order's reference it in the LoadAsync call below.

Adding the check for reference.CurrentValue == null fixed the problem, but so did setting the customer reference of the order to null in the test data setup. I've left in both fixes as I believe they are both right.